### PR TITLE
Better handling when empty instructions are unrolled or inverse

### PR DESCRIPTION
**Pull request by ajavadia**
handles some edge cases for empty circuits

### CODE REVIEW

File: qiskit/circuit/instruction.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def inverse(self):
             QiskitError: if the instruction is not composite
                 and an inverse has not been implemented for it.
         """
-        if not self.definition:
+        if self.definition is None:
```
Comment (ewinston) [inline qiskit/circuit/instruction.py]: why narrow this check? If an empty list comes along shouldn't that still raise this exception?

Comment (1ucian0) [inline qiskit/circuit/instruction.py]: What's the difference between `None` and empty in this context?

Comment (ajavadia) [reply qiskit/circuit/instruction.py]: `definition=[]` means an empty circuit:
```
q_0: |0>

q_1: |0>
```

`definition=None` means there is no definition for it and it is unknown how to expand it. i.e. opaque gate.

Comment (1ucian0) [reply qiskit/circuit/instruction.py]: gotcha...

**Pull request merged**